sample/app/build.gradle

```diff
@@ -28,6 +28,9 @@ buildProperties {
     env {
         from System.getenv()
     }
+    cli {
+        from project.properties
```

Comment:
As per [documentation](https://docs.gradle.org/3.3/userguide/build_environment.html#sec:gradle_configuration_properties):
> The configuration is applied in following order (if an option is configured in multiple locations the last one wins):
>
>* from gradle.properties in project build dir.
>* from gradle.properties in gradle user home.
>* from system properties, e.g. when -Dsome.property is set on the command line.

These 'build scopes' are combined with the dynamic scopes [provided by `Project`](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project.properties), which will override any value from the build scope above.